RealmSwift/AsymmetricObject.swift

```diff
@@ -0,0 +1,100 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2022 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import Foundation
+import Realm
+import Realm.Private
+
+/**
+ `AsymmetricObject` is a base class used to define asymmetric Realm objects.
+
+ Asymmetric objects can only be created using the `create(_ object:)`
+ function, and cannot be added, removed or queried.
+ When created, asymmetric objects will be synced unidirectionally to the MongoDB
+ database and cannot be accessed locally.
+
+ Incoming links from any asymmetric table are not allowed, meaning embedding
+ an asymmetric object within an `Object` will throw an error.
+
+ The property types supported on `AsymmetricObject` are the same as for `Object`,
+ except for that asymmetric objects can only link to embedded objects, so `Object`
+ and `List<Object>` properties are not supported (`EmbeddedObject` and
+ `List<EmbeddedObject>` *are*).
+
+ ```swift
+ class Person: AsymmetricObject {
+     @Persisted(primaryKey: true) var _id: ObjectId = ObjectId.generate()
+     @Persisted var name: String
+     @Persisted var age: Int
+ }
+ ```
+ */
+public typealias AsymmetricObject = RealmSwiftAsymmetricObject
+extension AsymmetricObject: _RealmCollectionValueInsideOptional {
```

Comment:
I don't think `_RealmCollectionValueInsideOptional` is correct? I may be missing something, but I don't think that AsymmetricObject should conform to `_Persistable` or `RealmCollectionValue`, as you can't link to an asymmetric object or have a collection of them.

Reply:
You are right, we are not supposed to use AsymmetricObjects within other top level object. Removing conformance to _RealmCollectionValueInsideOptional and _Persistable disallows using an asymmetric object within other Objects